cmd/node/node.go

```diff
@@ -661,7 +661,8 @@ func (app *App) initServices(ctx context.Context,
 		AlwaysListen:    app.Config.AlwaysListen,
 	}
 	syncer := syncer.NewSyncer(ctx, syncerConf, clock, msh, layerFetch, app.addLogger(SyncLogger, lg))
-
+	// TODO(dshulyak) this needs to be improved, but dependency graph is a bit complicated
+	tBeacon.SetSyncState(syncer)
```

Comment:
Why did you decide not to add this to the list of args to `New`?

Reply:
beacon needs to be created before syncer. there is a messy dependency order. but i can't resolve it without another layer of indirection